app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    result_args_dict = {}
+
+    def inner(*args) -> Callable:
```

Comment:
The return type of inner is not `Callable`, it would be a result of function calculations that could be `Any` by its type.